ethereumj-core/src/main/java/org/ethereum/util/RLP.java

```diff
@@ -1144,22 +1203,67 @@ else if (srcData.length == 1 && (srcData[0] & 0xFF) < 0x80) {
     }
 
 
-    private static int calculateLength(byte[] data, int index) {
-        if ((data[index] & 0xFF) >= OFFSET_LONG_ITEM
+    private static byte[] decodeItemBytes(byte[] data, int index) {
+
+        final int length = calculateItemLength(data, index);
+        // [0x80]
+        if (length == 0) {
+
+            return new byte[0];
+
+            // [0x00, 0x7f] - single byte with item
+        } else if ((data[index] & 0xFF) < OFFSET_SHORT_ITEM) {
+
+            byte[] valueBytes = new byte[1];
+            System.arraycopy(data, index, valueBytes, 0, 1);
+            return valueBytes;
+
+            // [0x01, 0xb7] - 1-55 bytes item
+        } else if ((data[index] & 0xFF) <= OFFSET_LONG_ITEM) {
+
+            byte[] valueBytes = new byte[length];
+            System.arraycopy(data, index+1, valueBytes, 0, length);
+            return valueBytes;
+
+            // [0xb8, 0xbf] - 56+ bytes item,
+            // for 0xc0 and greater exception is thrown before in {@link #calculateItemLength}
```

Comment:
Not sure that it is good practice to relay on exception thrown by another method. Somebody may not consider that fact and make changes that will break designed behavior.

Reply:
Fixed